scripts/add-new-lords

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+import re
+import urllib2
+from lxml import etree
+from popolo import PopoloPlus
+from popolo.utils import new_id
+
+parser = etree.ETCompatXMLParser()
+etree.set_default_parser(parser)
+
+data = PopoloPlus()
+
+parl_ids = {}
+for person in data.persons.values():
+    for i in person.get('identifiers', []):
+        if i['scheme'] == 'datadotparl_id':
+            parl_ids[person['id']] = i['identifier']
+
+
+def canon_name(n):
+    n = re.sub(' St\. +', ' St ', n)
+    n = re.sub('^The ', '', n)
+    n = re.sub(' De ', ' de ', n)
+    return n
+
+
+LORDS_URL = 'http://data.parliament.uk/membersdataplatform/services/mnis/members/query/house=Lords/BasicDetails'
+parl_members = etree.parse(urllib2.urlopen(LORDS_URL)).getroot()
+
+lord_match = re.compile('^([^\s]*)\s+.*\sof\s+(.*)$')
+
+party_map = {
+    'Liberal Democrats': 'liberal-democrat',
+    'Conservative': 'conservative',
+    'Crossbench': 'crossbench',
+    'Labour': 'labour',
+    'Bishops': 'bishop'
+}
+
+type_map = {
+    'Life peer': 'L',
+    'Excepted Hereditary': 'HP',
+    'Bishops': 'B'
+}
+
+new_people = False
+parl_by_name = {}
+for member in parl_members:
+    end_date = member.find('HouseEndDate').text or ''
+    end_date = end_date.replace('T00:00:00', '')
+    if end_date and end_date < '1999-11-12':
+        continue
+    name = canon_name(member.find('DisplayAs').text)
+    if not data.get_person(name=name):
+        if member.attrib['Pims_Id'] not in data.pims_ids:
+            print "{} is a new Lord".format(name)
+            person_id = new_id(data.max_person_id())
+            lord_id = new_id(data.max_lord_id())
+
+            given_name = member.find('BasicDetails/GivenForename').text
+            middle_names = member.find('BasicDetails/GivenMiddleNames').text or ''
+            names = u' '.join((given_name, middle_names)).strip()
+
+            surname = member.find('BasicDetails/GivenSurname').text
```

Comment:
Their surname might not be their lord name (Douglas Hogg is Viscount Hailsham), so we should use the lord name :)